packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/marker.dart

```diff
@@ -259,6 +283,55 @@ class Marker implements MapsObject<Marker> {
       onDragStart: onDragStartParam ?? onDragStart,
       onDrag: onDragParam ?? onDrag,
       onDragEnd: onDragEndParam ?? onDragEnd,
+      clusterManagerId: clusterManagerIdParam ?? clusterManagerId,
+    );
+  }
+
+  /// Creates a new [Marker] object whose values are the same as this instance,
+  /// unless overwritten by the default values for specified parameters.
+  Marker copyWithDefaults({
```

Comment:
What is the use case for this method? I've never seen this pattern.

Reply:
Marker objects are immutable objects. Only way to update marker is to create new one or by using markers copyWith functionality. 
Marker has some nullable values: `onTap`, `onDrag`, `onDragStart`, `onDragEnd`, `clusterManagerId`.
Current copyWith pattern does not allow clearing these values. For example if you want to remove marker from the cluster, following line won't work `marker.copyWith(clusterManagerId: null)` as clusterManagerId is not updated if given value is null: 
```dart
  Marker copyWith({
    ...
    ClusterManagerId? clusterManagerIdParam,
  }) {
    return Marker(
      ...
      clusterManagerId: clusterManagerIdParam ?? clusterManagerId,
    );
  }
```

copyWithDefaults allow copying the marker and using default values for selected fields:
`Marker copy = marker.copyWithDefaults(defaultOnDrag: true,defaultClusterManagerId);`

This is is not perfect pattern, but changing the copyWith behaviour would be breaking change.

One pattern could be less generic methods like `copyWithNullClusterManagerId()` if that sounds better option.

I personally needed this on tests, as I wanted to remove marker from clustermanager and check that this case works, but this might be quite rare case for developers. I can move this functionality as a helper method under tests, if that suits better.

Reply:
copyWithDefaults method is now removed and needed functionality (ability to remove marker from cluster) is moved to test files. This means that testing files need to be kept up with marker changes, but it affects only for these few test cases, and should not be too big task to do. 
In those rare cases where developers need to be able to set `onTap`, `onDrag`, `onDragStart`, `onDragEnd` or `clusterManagerId` to `null`, they are always able to code own implementation of the feature.